flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -930,6 +930,7 @@ public void serialize(NFA<T> record, DataOutputView target) throws IOException {
 				timestampSerializer.serialize(computationState.getTimestamp(), target);
 				versionSerializer.serialize(computationState.getVersion(), target);
 				timestampSerializer.serialize(computationState.getStartTimestamp(), target);
+				target.writeInt(computationState.getCounter());
```

Comment:
The `counter` field was introduced in Flink 1.3 and the new serializer is only used in Flink 1.3. Before it was Java Serialization, so if you see in the `readComputationState()` we create a computation state with 0.